lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/builtins/MethodGenerator.java

```diff
@@ -144,121 +142,113 @@ protected MethodParameter fromVariableElementToMethodParameter(
+    @Override
+    public List<SafeWrapException> visitArray(List<? extends AnnotationValue> vals, Object o) {
+      for (var annotationValue : vals) {
+        annotationValue.accept(this, o);
+      }
+      return elements;
+    }
 
-      List<SafeWrapException> wrappedExceptions = new ArrayList<>();
-      for (AnnotationMirror am : element.getAnnotationMirrors()) {
-        if (tpeUtils.isSameType(am.getAnnotationType(), builtinType)) {
-          for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
-              am.getElementValues().entrySet()) {
-            if (ValueElementName.equals(entry.getKey().getSimpleName().toString())) {
-              Attribute.Array wrapExceptions = (Attribute.Array) entry.getValue();
-              for (int i = 0; i < wrapExceptions.values.length; i++) {
-                Attribute.Class valueFrom = null;
-                Attribute.Class valueTo = null;
-                Attribute.Compound attr = (Attribute.Compound) wrapExceptions.values[i];
-                for (Pair<Symbol.MethodSymbol, Attribute> p : attr.values) {
-                  Name key = p.fst.getSimpleName();
-                  if (key.contentEquals(FromElementName)) {
-                    valueFrom = (Attribute.Class) p.snd;
-                  } else if (key.contentEquals(ToElementName)) {
-                    valueTo = (Attribute.Class) p.snd;
-                  }
-                }
-                if (valueFrom != null) {
-                  SafeWrapException converted =
-                      new SafeWrapException(valueFrom, Optional.ofNullable(valueTo));
-                  wrappedExceptions.add(converted);
-                }
-              }
-              break;
-            }
-          }
+    @Override
+    public List<SafeWrapException> visitAnnotation(AnnotationMirror a, Object o) {
+      TypeElement valueFrom = null;
+      TypeElement valueTo = null;
+      for (var entry : a.getElementValues().entrySet()) {
+        var name = entry.getKey().getSimpleName().toString();
+        switch (name) {
+          case FROM_ELEMENT_NAME -> valueFrom = entry.getValue().accept(typeVisitor, null);
+          case TO_ELEMENT_NAME -> valueTo = entry.getValue().accept(typeVisitor, null);
+          default -> processingEnvironment.getMessager().printMessage(
+              Kind.ERROR,
+              "Unknown annotation element name: " + name);
         }
       }
-      return wrappedExceptions.toArray(new SafeWrapException[0]);
+      if (valueFrom != null) {
+        var safeWrapException = new SafeWrapException(valueFrom, Optional.ofNullable(valueTo));
+        elements.add(safeWrapException);
+      }
+      return elements;
+    }
+  }
+
+  private class AnnotationTypeVisitor extends SimpleAnnotationValueVisitor14<TypeElement, Object> {
+    @Override
+    public TypeElement visitType(TypeMirror t, Object o) {
+      var typeElement = processingEnvironment.getElementUtils().getTypeElement(t.toString());
```

Comment:
There should be a way to convert `TypeMirror` to element. I think it is: [Types.asElement](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/util/Types.html#asElement-javax.lang.model.type.TypeMirror-)

Reply:
Turns out there is a way to do that with [ElementVisitor](https://docs.oracle.com/javase/8/docs/api/javax/lang/model/element/ElementVisitor.html).

Reply:
Fixed in 1cf518556b73a6fa082625844dcafb78b92a558f

Reply:
Visitor over visitor, but that's how the JDK guys want it to be! No more complains, this is the nicest annotation processor code I can imagine.